src/Stratis.SmartContracts.IntegrationTests/PoW/SmartContractWalletTests.cs

```diff
@@ -437,6 +437,63 @@ public void MockChain_AuctionTest()
             }
         }
 
+        [Fact]
+        public void MockChain_Lottery()
```

Comment:
Do we need to add this to PoA tests too?

Reply:
Good point, I only did it here because it's easier - I'll add it to this PR after hopefully the PoA change is merged in